Content.Shared/PDA/Ringer/RingerComponent.cs

```diff
@@ -0,0 +1,57 @@
+using Robust.Shared.GameStates;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom;
+
+namespace Content.Shared.PDA.Ringer;
+
+[RegisterComponent, NetworkedComponent, Access(typeof(SharedRingerSystem))]
+[AutoGenerateComponentState(true, true), AutoGenerateComponentPause]
+public sealed partial class RingerComponent : Component
+{
+    /// <summary>
+    /// The ringtone, represented as an array of notes.
+    /// </summary>
+    [DataField, AutoNetworkedField]
+    public Note[] Ringtone = new Note[SharedRingerSystem.RingtoneLength];
+
+    /// <summary>
+    /// The last time this ringer's ringtone was set.
+    /// </summary>
+    [DataField(customTypeSerializer: typeof(TimeOffsetSerializer)), AutoPausedField, AutoNetworkedField]
+    public TimeSpan LastRingtoneSetTime;
+
```

Comment:
Treating it as NextRingtoneSetTime means you don't need to do last time + delay every single time on it.